tuf/api/metadata.py

```diff
@@ -489,38 +531,39 @@ def __init__(
         version: int,
         spec_version: str,
         expires: datetime,
-        meta: Mapping[str, Any],
+        meta: Mapping[str, MetadataInfo],
     ) -> None:
         super().__init__(_type, version, spec_version, expires)
-        # TODO: Add class for meta
         self.meta = meta
 
     @classmethod
     def from_dict(cls, timestamp_dict: Mapping[str, Any]) -> "Timestamp":
         """Creates Timestamp object from its dict representation. """
         common_args = cls._common_fields_from_dict(timestamp_dict)
-        meta = timestamp_dict.pop("meta")
+        meta = {
+            "snapshot.json": MetadataInfo(
```

Comment:
Same comment as above, related to hardcoding file formats. In this case if an implementation does not use `.json` files this will probably raise a `KeyError`.
Possibly the filename/format should be read from `timestamp_dict` instead.

Reply:
That's a really good catch!
Will make it work with other extensions as well!